api/v1alpha3/machine_webhook.go

```diff
@@ -40,6 +40,8 @@ func (m *Machine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 var _ webhook.Validator = &Machine{}
 var _ webhook.Defaulter = &Machine{}
 
+var kubeSemver = regexp.MustCompile(`^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?$`)
```

Comment:
If we are doing the defaulting/validation in the webhook, any particular reason to use the regex rather than semver parsing to validate?

Reply:
I assume because the original intent was to match what KCP is doing today

Reply:
Yes, that, and also because the `v` prefix is required now. And that way we could easily move the regex to schema validation in v1alpha4 instead of validating in the webhook.

Reply:
gotcha, sgtm, thanks for the clarification.